utils/threshAlgos.py

```diff
@@ -625,10 +637,43 @@ def calibrateThrDAC(inputFile,fitRange="0,255",listOfVFATs=None,noLeg=False,outp
                 dict_ScurveSigmaVsThrDac_BoxPlot[vfat].SetXTitle(thrDacName)
                 dict_ScurveSigmaVsThrDac_BoxPlot[vfat].SetYTitle("Scurve Sigma #left(fC#right)")
 
+            if vfat > -1:    
+                scurveFitDataThisVfat = scurveFitData[scurveFitData["vfatN"] == vfat]
+            else:
+                scurveFitDataThisVfat = scurveFitData
+
+            if len(scurveFitDataThisVfat) == 0:
+                thisVFAT_ThreshMean = 0
+                thisVFAT_ThreshStd = 1
+                histThresh = r.TH1F("scurveMean_vfat%i"%vfat,"VFAT %i;S-Curve Mean #left(fC#right);N"%vfat, 
+                                    40, thisVFAT_ThreshMean - 5. * thisVFAT_ThreshStd, thisVFAT_ThreshMean + 5. * thisVFAT_ThreshStd )
+
+                thisVFAT_ENCMean = 0
+                thisVFAT_ENCStd = 1
+                histENC = r.TH1F("scurveSigma_vfat%i"%vfat,"VFAT %i;S-Curve Sigma #left(fC#right);N"%vfat,
+                                 40, thisVFAT_ENCMean - 5. * thisVFAT_ENCStd, thisVFAT_ENCMean + 5. * thisVFAT_ENCStd )                
+            else:    
+                thisVFAT_ThreshMean = np.mean(scurveFitDataThisVfat["threshold"])
+                thisVFAT_ThreshStd = np.std(scurveFitDataThisVfat["threshold"])
+                histThresh = r.TH1F("scurveMean_vfat%i"%vfat,"VFAT %i;S-Curve Mean #left(fC#right);N"%vfat, 
+                                    40, thisVFAT_ThreshMean - 5. * thisVFAT_ThreshStd, thisVFAT_ThreshMean + 5. * thisVFAT_ThreshStd )
+                
+                thisVFAT_ENCMean = np.mean(scurveFitDataThisVfat["noise"])
+                thisVFAT_ENCStd = np.std(scurveFitDataThisVfat["noise"])
+                histENC = r.TH1F("scurveSigma_vfat%i"%vfat,"VFAT %i;S-Curve Sigma #left(fC#right);N"%vfat,
+                                 40, thisVFAT_ENCMean - 5. * thisVFAT_ENCStd, thisVFAT_ENCMean + 5. * thisVFAT_ENCStd )
+        
+            for idy in range(0,len(scurveFitDataThisVfat)):
+                scurveMean = scurveFitDataThisVfat[idy]['threshold']
+                scurveSigma = scurveFitDataThisVfat[idy]['noise']
+                histThresh.Fill(scurveMean)
+                histENC.Fill(scurveSigma)
```

Comment:
What is it that you're adding with this block? It wasn't discussed in the issue.

Reply:
We are loading data into the ARM DAC calibration script from the TTree, instead of the TGraphs because we need to apply the quality cuts.

Reply:
As I mentioned in your follow-up comment on the github issue it would be good to add some comments here to "guide" the reader.

Reply:
code streamlined and more comments added